app/components/primer/auto_complete_component.rb

```diff
@@ -0,0 +1,97 @@
+
+    # Optional icon to be rendered before the input. Has the same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
+    renders_one :icon, Primer::OcticonComponent
+
+    # Customizable results list.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :results, lambda { |**system_arguments|
+      system_arguments[:tag] = :ul
+      system_arguments[:id] = @id
+      system_arguments[:classes] = class_names(
+        "autocomplete-results",
+        system_arguments[:classes]
+      )
+
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # @example Default
+    #   <%= render(Primer::AutoCompleteComponent.new(src: "/users/search", id: "user-popup", position: :relative)) do |c| %>
+    #     <% c.input(type: :text, name: "input", classes: "form-control") %>
+    #     <% c.results do %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(selected: true, value: "value")) do |c| %>
+    #         Selected
+    #       <% end %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(value: "value")) do |c| %>
+    #         Not selected
+    #       <% end %>
+    #     <% end %>
+    #   <% end %>
+    #
+    # @example With custom classes for the results
+    #   <%= render(Primer::AutoCompleteComponent.new(src: "/users/search", id: "user-popup", position: :relative)) do |c| %>
+    #     <% c.input(type: :text, name: "input", classes: "form-control") %>
+    #     <% c.results(classes: "my-custom-class") do %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(selected: true, value: "value")) do |c| %>
+    #         Selected
+    #       <% end %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(value: "value")) do |c| %>
+    #         Not selected
+    #       <% end %>
+    #     <% end %>
+    #   <% end %>
+    #
+    # @example With Icon
+    #   <%= render(Primer::AutoCompleteComponent.new(src: "/users/search", id: "user-popup", position: :relative)) do |c| %>
+    #     <% c.input(type: :text, name: "input", classes: "form-control") %>
+    #     <% c.icon(icon: :search) %>
+    #     <% c.results do %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(selected: true, value: "value")) do |c| %>
+    #         Selected
+    #       <% end %>
+    #       <%= render(Primer::AutoCompleteItemComponent.new(value: "value")) do |c| %>
+    #         Not selected
+    #       <% end %>
+    #     <% end %>
+    #   <% end %>
+    #
+    # @param src [String] The route to query.
+    # @param id [String] Id of the list element.
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(src:, id:, **system_arguments)
+      @id = id
+
+      @system_arguments = system_arguments
+      @system_arguments[:tag] = "auto-complete"
+      @system_arguments[:src] = src
+      @system_arguments[:for] = id
+    end
+
+    # add `results` without needing to explicitly call it in the view
```

Comment:
Could you elaborate on what this does? It surprised me a bit.

Reply:
the idea is that the user does not need to manually call `c.results` on the view

so
```erb
<%= render(Primer::AutoCompleteComponent.new) %>
```

and 

```erb
<%= render(Primer::AutoCompleteComponent.new)  do |c| %>
  <% c.result() %>
<% end %>
```
 are the same thing. ideally, the user would only call `results` if they want to customize it